// @aws-sdk/client-acm/dist-es/commands/ExportCertificateCommand.js
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { getSerdePlugin } from "@smithy/middleware-serde";
import { Command as $Command } from "@smithy/smithy-client";
import { SMITHY_CONTEXT_KEY, } from "@smithy/types";
import { ExportCertificateRequestFilterSensitiveLog, ExportCertificateResponseFilterSensitiveLog, } from "../models/models_0";
import { de_ExportCertificateCommand, se_ExportCertificateCommand } from "../protocols/Aws_json1_1";
export { $Command };
export class ExportCertificateCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ExportCertificateCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ACMClient";
        const commandName = "ExportCertificateCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ExportCertificateRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ExportCertificateResponseFilterSensitiveLog,
            [SMITHY_CONTEXT_KEY]: {
                service: "CertificateManager",
                operation: "ExportCertificate",
            },
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_ExportCertificateCommand(input, context);
    }
    deserialize(output, context) {
        return de_ExportCertificateCommand(output, context);
    }
}
