// @aws-sdk/client-api-gateway/dist-cjs/APIGateway.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIGateway = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const APIGatewayClient_1 = require("./APIGatewayClient");
const CreateApiKeyCommand_1 = require("./commands/CreateApiKeyCommand");
const CreateAuthorizerCommand_1 = require("./commands/CreateAuthorizerCommand");
const CreateBasePathMappingCommand_1 = require("./commands/CreateBasePathMappingCommand");
const CreateDeploymentCommand_1 = require("./commands/CreateDeploymentCommand");
const CreateDocumentationPartCommand_1 = require("./commands/CreateDocumentationPartCommand");
const CreateDocumentationVersionCommand_1 = require("./commands/CreateDocumentationVersionCommand");
const CreateDomainNameCommand_1 = require("./commands/CreateDomainNameCommand");
const CreateModelCommand_1 = require("./commands/CreateModelCommand");
const CreateRequestValidatorCommand_1 = require("./commands/CreateRequestValidatorCommand");
const CreateResourceCommand_1 = require("./commands/CreateResourceCommand");
const CreateRestApiCommand_1 = require("./commands/CreateRestApiCommand");
const CreateStageCommand_1 = require("./commands/CreateStageCommand");
const CreateUsagePlanCommand_1 = require("./commands/CreateUsagePlanCommand");
const CreateUsagePlanKeyCommand_1 = require("./commands/CreateUsagePlanKeyCommand");
const CreateVpcLinkCommand_1 = require("./commands/CreateVpcLinkCommand");
const DeleteApiKeyCommand_1 = require("./commands/DeleteApiKeyCommand");
const DeleteAuthorizerCommand_1 = require("./commands/DeleteAuthorizerCommand");
const DeleteBasePathMappingCommand_1 = require("./commands/DeleteBasePathMappingCommand");
const DeleteClientCertificateCommand_1 = require("./commands/DeleteClientCertificateCommand");
const DeleteDeploymentCommand_1 = require("./commands/DeleteDeploymentCommand");
const DeleteDocumentationPartCommand_1 = require("./commands/DeleteDocumentationPartCommand");
const DeleteDocumentationVersionCommand_1 = require("./commands/DeleteDocumentationVersionCommand");
const DeleteDomainNameCommand_1 = require("./commands/DeleteDomainNameCommand");
const DeleteGatewayResponseCommand_1 = require("./commands/DeleteGatewayResponseCommand");
const DeleteIntegrationCommand_1 = require("./commands/DeleteIntegrationCommand");
const DeleteIntegrationResponseCommand_1 = require("./commands/DeleteIntegrationResponseCommand");
const DeleteMethodCommand_1 = require("./commands/DeleteMethodCommand");
const DeleteMethodResponseCommand_1 = require("./commands/DeleteMethodResponseCommand");
const DeleteModelCommand_1 = require("./commands/DeleteModelCommand");
const DeleteRequestValidatorCommand_1 = require("./commands/DeleteRequestValidatorCommand");
const DeleteResourceCommand_1 = require("./commands/DeleteResourceCommand");
const DeleteRestApiCommand_1 = require("./commands/DeleteRestApiCommand");
const DeleteStageCommand_1 = require("./commands/DeleteStageCommand");
const DeleteUsagePlanCommand_1 = require("./commands/DeleteUsagePlanCommand");
const DeleteUsagePlanKeyCommand_1 = require("./commands/DeleteUsagePlanKeyCommand");
const DeleteVpcLinkCommand_1 = require("./commands/DeleteVpcLinkCommand");
const FlushStageAuthorizersCacheCommand_1 = require("./commands/FlushStageAuthorizersCacheCommand");
const FlushStageCacheCommand_1 = require("./commands/FlushStageCacheCommand");
const GenerateClientCertificateCommand_1 = require("./commands/GenerateClientCertificateCommand");
const GetAccountCommand_1 = require("./commands/GetAccountCommand");
const GetApiKeyCommand_1 = require("./commands/GetApiKeyCommand");
const GetApiKeysCommand_1 = require("./commands/GetApiKeysCommand");
const GetAuthorizerCommand_1 = require("./commands/GetAuthorizerCommand");
const GetAuthorizersCommand_1 = require("./commands/GetAuthorizersCommand");
const GetBasePathMappingCommand_1 = require("./commands/GetBasePathMappingCommand");
const GetBasePathMappingsCommand_1 = require("./commands/GetBasePathMappingsCommand");
const GetClientCertificateCommand_1 = require("./commands/GetClientCertificateCommand");
const GetClientCertificatesCommand_1 = require("./commands/GetClientCertificatesCommand");
const GetDeploymentCommand_1 = require("./commands/GetDeploymentCommand");
const GetDeploymentsCommand_1 = require("./commands/GetDeploymentsCommand");
const GetDocumentationPartCommand_1 = require("./commands/GetDocumentationPartCommand");
const GetDocumentationPartsCommand_1 = require("./commands/GetDocumentationPartsCommand");
const GetDocumentationVersionCommand_1 = require("./commands/GetDocumentationVersionCommand");
const GetDocumentationVersionsCommand_1 = require("./commands/GetDocumentationVersionsCommand");
const GetDomainNameCommand_1 = require("./commands/GetDomainNameCommand");
const GetDomainNamesCommand_1 = require("./commands/GetDomainNamesCommand");
const GetExportCommand_1 = require("./commands/GetExportCommand");
const GetGatewayResponseCommand_1 = require("./commands/GetGatewayResponseCommand");
const GetGatewayResponsesCommand_1 = require("./commands/GetGatewayResponsesCommand");
const GetIntegrationCommand_1 = require("./commands/GetIntegrationCommand");
const GetIntegrationResponseCommand_1 = require("./commands/GetIntegrationResponseCommand");
const GetMethodCommand_1 = require("./commands/GetMethodCommand");
const GetMethodResponseCommand_1 = require("./commands/GetMethodResponseCommand");
const GetModelCommand_1 = require("./commands/GetModelCommand");
const GetModelsCommand_1 = require("./commands/GetModelsCommand");
const GetModelTemplateCommand_1 = require("./commands/GetModelTemplateCommand");
const GetRequestValidatorCommand_1 = require("./commands/GetRequestValidatorCommand");
const GetRequestValidatorsCommand_1 = require("./commands/GetRequestValidatorsCommand");
const GetResourceCommand_1 = require("./commands/GetResourceCommand");
const GetResourcesCommand_1 = require("./commands/GetResourcesCommand");
const GetRestApiCommand_1 = require("./commands/GetRestApiCommand");
const GetRestApisCommand_1 = require("./commands/GetRestApisCommand");
const GetSdkCommand_1 = require("./commands/GetSdkCommand");
const GetSdkTypeCommand_1 = require("./commands/GetSdkTypeCommand");
const GetSdkTypesCommand_1 = require("./commands/GetSdkTypesCommand");
const GetStageCommand_1 = require("./commands/GetStageCommand");
const GetStagesCommand_1 = require("./commands/GetStagesCommand");
const GetTagsCommand_1 = require("./commands/GetTagsCommand");
const GetUsageCommand_1 = require("./commands/GetUsageCommand");
const GetUsagePlanCommand_1 = require("./commands/GetUsagePlanCommand");
const GetUsagePlanKeyCommand_1 = require("./commands/GetUsagePlanKeyCommand");
const GetUsagePlanKeysCommand_1 = require("./commands/GetUsagePlanKeysCommand");
const GetUsagePlansCommand_1 = require("./commands/GetUsagePlansCommand");
const GetVpcLinkCommand_1 = require("./commands/GetVpcLinkCommand");
const GetVpcLinksCommand_1 = require("./commands/GetVpcLinksCommand");
const ImportApiKeysCommand_1 = require("./commands/ImportApiKeysCommand");
const ImportDocumentationPartsCommand_1 = require("./commands/ImportDocumentationPartsCommand");
const ImportRestApiCommand_1 = require("./commands/ImportRestApiCommand");
const PutGatewayResponseCommand_1 = require("./commands/PutGatewayResponseCommand");
const PutIntegrationCommand_1 = require("./commands/PutIntegrationCommand");
const PutIntegrationResponseCommand_1 = require("./commands/PutIntegrationResponseCommand");
const PutMethodCommand_1 = require("./commands/PutMethodCommand");
const PutMethodResponseCommand_1 = require("./commands/PutMethodResponseCommand");
const PutRestApiCommand_1 = require("./commands/PutRestApiCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const TestInvokeAuthorizerCommand_1 = require("./commands/TestInvokeAuthorizerCommand");
const TestInvokeMethodCommand_1 = require("./commands/TestInvokeMethodCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateAccountCommand_1 = require("./commands/UpdateAccountCommand");
const UpdateApiKeyCommand_1 = require("./commands/UpdateApiKeyCommand");
const UpdateAuthorizerCommand_1 = require("./commands/UpdateAuthorizerCommand");
const UpdateBasePathMappingCommand_1 = require("./commands/UpdateBasePathMappingCommand");
const UpdateClientCertificateCommand_1 = require("./commands/UpdateClientCertificateCommand");
const UpdateDeploymentCommand_1 = require("./commands/UpdateDeploymentCommand");
const UpdateDocumentationPartCommand_1 = require("./commands/UpdateDocumentationPartCommand");
const UpdateDocumentationVersionCommand_1 = require("./commands/UpdateDocumentationVersionCommand");
const UpdateDomainNameCommand_1 = require("./commands/UpdateDomainNameCommand");
const UpdateGatewayResponseCommand_1 = require("./commands/UpdateGatewayResponseCommand");
const UpdateIntegrationCommand_1 = require("./commands/UpdateIntegrationCommand");
const UpdateIntegrationResponseCommand_1 = require("./commands/UpdateIntegrationResponseCommand");
const UpdateMethodCommand_1 = require("./commands/UpdateMethodCommand");
const UpdateMethodResponseCommand_1 = require("./commands/UpdateMethodResponseCommand");
const UpdateModelCommand_1 = require("./commands/UpdateModelCommand");
const UpdateRequestValidatorCommand_1 = require("./commands/UpdateRequestValidatorCommand");
const UpdateResourceCommand_1 = require("./commands/UpdateResourceCommand");
const UpdateRestApiCommand_1 = require("./commands/UpdateRestApiCommand");
const UpdateStageCommand_1 = require("./commands/UpdateStageCommand");
const UpdateUsageCommand_1 = require("./commands/UpdateUsageCommand");
const UpdateUsagePlanCommand_1 = require("./commands/UpdateUsagePlanCommand");
const UpdateVpcLinkCommand_1 = require("./commands/UpdateVpcLinkCommand");
const commands = {
    CreateApiKeyCommand: CreateApiKeyCommand_1.CreateApiKeyCommand,
    CreateAuthorizerCommand: CreateAuthorizerCommand_1.CreateAuthorizerCommand,
    CreateBasePathMappingCommand: CreateBasePathMappingCommand_1.CreateBasePathMappingCommand,
    CreateDeploymentCommand: CreateDeploymentCommand_1.CreateDeploymentCommand,
    CreateDocumentationPartCommand: CreateDocumentationPartCommand_1.CreateDocumentationPartCommand,
    CreateDocumentationVersionCommand: CreateDocumentationVersionCommand_1.CreateDocumentationVersionCommand,
    CreateDomainNameCommand: CreateDomainNameCommand_1.CreateDomainNameCommand,
    CreateModelCommand: CreateModelCommand_1.CreateModelCommand,
    CreateRequestValidatorCommand: CreateRequestValidatorCommand_1.CreateRequestValidatorCommand,
    CreateResourceCommand: CreateResourceCommand_1.CreateResourceCommand,
    CreateRestApiCommand: CreateRestApiCommand_1.CreateRestApiCommand,
    CreateStageCommand: CreateStageCommand_1.CreateStageCommand,
    CreateUsagePlanCommand: CreateUsagePlanCommand_1.CreateUsagePlanCommand,
    CreateUsagePlanKeyCommand: CreateUsagePlanKeyCommand_1.CreateUsagePlanKeyCommand,
    CreateVpcLinkCommand: CreateVpcLinkCommand_1.CreateVpcLinkCommand,
    DeleteApiKeyCommand: DeleteApiKeyCommand_1.DeleteApiKeyCommand,
    DeleteAuthorizerCommand: DeleteAuthorizerCommand_1.DeleteAuthorizerCommand,
    DeleteBasePathMappingCommand: DeleteBasePathMappingCommand_1.DeleteBasePathMappingCommand,
    DeleteClientCertificateCommand: DeleteClientCertificateCommand_1.DeleteClientCertificateCommand,
    DeleteDeploymentCommand: DeleteDeploymentCommand_1.DeleteDeploymentCommand,
    DeleteDocumentationPartCommand: DeleteDocumentationPartCommand_1.DeleteDocumentationPartCommand,
    DeleteDocumentationVersionCommand: DeleteDocumentationVersionCommand_1.DeleteDocumentationVersionCommand,
    DeleteDomainNameCommand: DeleteDomainNameCommand_1.DeleteDomainNameCommand,
    DeleteGatewayResponseCommand: DeleteGatewayResponseCommand_1.DeleteGatewayResponseCommand,
    DeleteIntegrationCommand: DeleteIntegrationCommand_1.DeleteIntegrationCommand,
    DeleteIntegrationResponseCommand: DeleteIntegrationResponseCommand_1.DeleteIntegrationResponseCommand,
    DeleteMethodCommand: DeleteMethodCommand_1.DeleteMethodCommand,
    DeleteMethodResponseCommand: DeleteMethodResponseCommand_1.DeleteMethodResponseCommand,
    DeleteModelCommand: DeleteModelCommand_1.DeleteModelCommand,
    DeleteRequestValidatorCommand: DeleteRequestValidatorCommand_1.DeleteRequestValidatorCommand,
    DeleteResourceCommand: DeleteResourceCommand_1.DeleteResourceCommand,
    DeleteRestApiCommand: DeleteRestApiCommand_1.DeleteRestApiCommand,
    DeleteStageCommand: DeleteStageCommand_1.DeleteStageCommand,
    DeleteUsagePlanCommand: DeleteUsagePlanCommand_1.DeleteUsagePlanCommand,
    DeleteUsagePlanKeyCommand: DeleteUsagePlanKeyCommand_1.DeleteUsagePlanKeyCommand,
    DeleteVpcLinkCommand: DeleteVpcLinkCommand_1.DeleteVpcLinkCommand,
    FlushStageAuthorizersCacheCommand: FlushStageAuthorizersCacheCommand_1.FlushStageAuthorizersCacheCommand,
    FlushStageCacheCommand: FlushStageCacheCommand_1.FlushStageCacheCommand,
    GenerateClientCertificateCommand: GenerateClientCertificateCommand_1.GenerateClientCertificateCommand,
    GetAccountCommand: GetAccountCommand_1.GetAccountCommand,
    GetApiKeyCommand: GetApiKeyCommand_1.GetApiKeyCommand,
    GetApiKeysCommand: GetApiKeysCommand_1.GetApiKeysCommand,
    GetAuthorizerCommand: GetAuthorizerCommand_1.GetAuthorizerCommand,
    GetAuthorizersCommand: GetAuthorizersCommand_1.GetAuthorizersCommand,
    GetBasePathMappingCommand: GetBasePathMappingCommand_1.GetBasePathMappingCommand,
    GetBasePathMappingsCommand: GetBasePathMappingsCommand_1.GetBasePathMappingsCommand,
    GetClientCertificateCommand: GetClientCertificateCommand_1.GetClientCertificateCommand,
    GetClientCertificatesCommand: GetClientCertificatesCommand_1.GetClientCertificatesCommand,
    GetDeploymentCommand: GetDeploymentCommand_1.GetDeploymentCommand,
    GetDeploymentsCommand: GetDeploymentsCommand_1.GetDeploymentsCommand,
    GetDocumentationPartCommand: GetDocumentationPartCommand_1.GetDocumentationPartCommand,
    GetDocumentationPartsCommand: GetDocumentationPartsCommand_1.GetDocumentationPartsCommand,
    GetDocumentationVersionCommand: GetDocumentationVersionCommand_1.GetDocumentationVersionCommand,
    GetDocumentationVersionsCommand: GetDocumentationVersionsCommand_1.GetDocumentationVersionsCommand,
    GetDomainNameCommand: GetDomainNameCommand_1.GetDomainNameCommand,
    GetDomainNamesCommand: GetDomainNamesCommand_1.GetDomainNamesCommand,
    GetExportCommand: GetExportCommand_1.GetExportCommand,
    GetGatewayResponseCommand: GetGatewayResponseCommand_1.GetGatewayResponseCommand,
    GetGatewayResponsesCommand: GetGatewayResponsesCommand_1.GetGatewayResponsesCommand,
    GetIntegrationCommand: GetIntegrationCommand_1.GetIntegrationCommand,
    GetIntegrationResponseCommand: GetIntegrationResponseCommand_1.GetIntegrationResponseCommand,
    GetMethodCommand: GetMethodCommand_1.GetMethodCommand,
    GetMethodResponseCommand: GetMethodResponseCommand_1.GetMethodResponseCommand,
    GetModelCommand: GetModelCommand_1.GetModelCommand,
    GetModelsCommand: GetModelsCommand_1.GetModelsCommand,
    GetModelTemplateCommand: GetModelTemplateCommand_1.GetModelTemplateCommand,
    GetRequestValidatorCommand: GetRequestValidatorCommand_1.GetRequestValidatorCommand,
    GetRequestValidatorsCommand: GetRequestValidatorsCommand_1.GetRequestValidatorsCommand,
    GetResourceCommand: GetResourceCommand_1.GetResourceCommand,
    GetResourcesCommand: GetResourcesCommand_1.GetResourcesCommand,
    GetRestApiCommand: GetRestApiCommand_1.GetRestApiCommand,
    GetRestApisCommand: GetRestApisCommand_1.GetRestApisCommand,
    GetSdkCommand: GetSdkCommand_1.GetSdkCommand,
    GetSdkTypeCommand: GetSdkTypeCommand_1.GetSdkTypeCommand,
    GetSdkTypesCommand: GetSdkTypesCommand_1.GetSdkTypesCommand,
    GetStageCommand: GetStageCommand_1.GetStageCommand,
    GetStagesCommand: GetStagesCommand_1.GetStagesCommand,
    GetTagsCommand: GetTagsCommand_1.GetTagsCommand,
    GetUsageCommand: GetUsageCommand_1.GetUsageCommand,
    GetUsagePlanCommand: GetUsagePlanCommand_1.GetUsagePlanCommand,
    GetUsagePlanKeyCommand: GetUsagePlanKeyCommand_1.GetUsagePlanKeyCommand,
    GetUsagePlanKeysCommand: GetUsagePlanKeysCommand_1.GetUsagePlanKeysCommand,
    GetUsagePlansCommand: GetUsagePlansCommand_1.GetUsagePlansCommand,
    GetVpcLinkCommand: GetVpcLinkCommand_1.GetVpcLinkCommand,
    GetVpcLinksCommand: GetVpcLinksCommand_1.GetVpcLinksCommand,
    ImportApiKeysCommand: ImportApiKeysCommand_1.ImportApiKeysCommand,
    ImportDocumentationPartsCommand: ImportDocumentationPartsCommand_1.ImportDocumentationPartsCommand,
    ImportRestApiCommand: ImportRestApiCommand_1.ImportRestApiCommand,
    PutGatewayResponseCommand: PutGatewayResponseCommand_1.PutGatewayResponseCommand,
    PutIntegrationCommand: PutIntegrationCommand_1.PutIntegrationCommand,
    PutIntegrationResponseCommand: PutIntegrationResponseCommand_1.PutIntegrationResponseCommand,
    PutMethodCommand: PutMethodCommand_1.PutMethodCommand,
    PutMethodResponseCommand: PutMethodResponseCommand_1.PutMethodResponseCommand,
    PutRestApiCommand: PutRestApiCommand_1.PutRestApiCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    TestInvokeAuthorizerCommand: TestInvokeAuthorizerCommand_1.TestInvokeAuthorizerCommand,
    TestInvokeMethodCommand: TestInvokeMethodCommand_1.TestInvokeMethodCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateAccountCommand: UpdateAccountCommand_1.UpdateAccountCommand,
    UpdateApiKeyCommand: UpdateApiKeyCommand_1.UpdateApiKeyCommand,
    UpdateAuthorizerCommand: UpdateAuthorizerCommand_1.UpdateAuthorizerCommand,
    UpdateBasePathMappingCommand: UpdateBasePathMappingCommand_1.UpdateBasePathMappingCommand,
    UpdateClientCertificateCommand: UpdateClientCertificateCommand_1.UpdateClientCertificateCommand,
    UpdateDeploymentCommand: UpdateDeploymentCommand_1.UpdateDeploymentCommand,
    UpdateDocumentationPartCommand: UpdateDocumentationPartCommand_1.UpdateDocumentationPartCommand,
    UpdateDocumentationVersionCommand: UpdateDocumentationVersionCommand_1.UpdateDocumentationVersionCommand,
    UpdateDomainNameCommand: UpdateDomainNameCommand_1.UpdateDomainNameCommand,
    UpdateGatewayResponseCommand: UpdateGatewayResponseCommand_1.UpdateGatewayResponseCommand,
    UpdateIntegrationCommand: UpdateIntegrationCommand_1.UpdateIntegrationCommand,
    UpdateIntegrationResponseCommand: UpdateIntegrationResponseCommand_1.UpdateIntegrationResponseCommand,
    UpdateMethodCommand: UpdateMethodCommand_1.UpdateMethodCommand,
    UpdateMethodResponseCommand: UpdateMethodResponseCommand_1.UpdateMethodResponseCommand,
    UpdateModelCommand: UpdateModelCommand_1.UpdateModelCommand,
    UpdateRequestValidatorCommand: UpdateRequestValidatorCommand_1.UpdateRequestValidatorCommand,
    UpdateResourceCommand: UpdateResourceCommand_1.UpdateResourceCommand,
    UpdateRestApiCommand: UpdateRestApiCommand_1.UpdateRestApiCommand,
    UpdateStageCommand: UpdateStageCommand_1.UpdateStageCommand,
    UpdateUsageCommand: UpdateUsageCommand_1.UpdateUsageCommand,
    UpdateUsagePlanCommand: UpdateUsagePlanCommand_1.UpdateUsagePlanCommand,
    UpdateVpcLinkCommand: UpdateVpcLinkCommand_1.UpdateVpcLinkCommand,
};
class APIGateway extends APIGatewayClient_1.APIGatewayClient {
}
exports.APIGateway = APIGateway;
(0, smithy_client_1.createAggregatedClient)(commands, APIGateway);
